pkgs/applications/video/aegisub/default.nix

```diff
@@ -60,58 +70,84 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "wangqr";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-Er0g8fJyx7zjNVpKw7zUHE40hU10BdYlZohlqJq2LE0=";
+    rev = "91f8b5f91eb960bad19899c10af08aca34f9b697";
+    sha256 = "sha256-lPkPWSsncsBKJHDnma9cUXsQJynruT9JpPkMTHdQ/e8=";
   };
 
-  patches = [ ./no-git.patch ];
-
   nativeBuildInputs = [
-    autoreconfHook
     intltool
     luajit52
     pkg-config
     which
+    cmake
   ];
+
   buildInputs = [
     boost
+    expat
     ffmpeg
     ffms
     fftw
     fontconfig
     freetype
+    fribidi
+    glib
+    harfbuzz
     icu
     libGL
     libGLU
     libX11
     libass
     libiconv
     libuchardet
+    pcre
     wxGTK
     zlib
   ]
+  ++ lib.optionals stdenv.isDarwin [
+    CoreText
+    CoreFoundation
+    AppKit
+    Carbon
+    IOKit
+    Cocoa
+  ]
   ++ optional alsaSupport alsa-lib
-  ++ optional automationSupport luajit52
   ++ optional openalSupport openal
   ++ optional portaudioSupport portaudio
   ++ optional pulseaudioSupport libpulseaudio
   ++ optional spellcheckSupport hunspell
   ;
 
-  configureFlags = [
-    "--with-boost-libdir=${boost.out}/lib"
-    "--with-system-luajit"
-    "FORCE_GIT_VERSION=${version}"
-  ];
-
   enableParallelBuilding = true;
 
   hardeningDisable = [
     "bindnow"
     "relro"
   ];
 
-  postInstall = "ln -s $out/bin/aegisub-* $out/bin/aegisub";
+  patchPhase = lib.optionalString (!useBundledLuaJIT) ''
+    substituteInPlace CMakeLists.txt \
+      --replace "luajit)" "luajit-5.1)" \
+      --replace "luajit " "luajit-5.1 " \
+      --replace "luajit-minilua" "luajit"
+    substituteInPlace CMakeLists.test.txt \
+      --replace "luajit " "luajit-5.1 "
+    substituteInPlace vendor/luabins/CMakeLists.txt \
+      --replace "luajit)" "luajit-5.1)"
+    sed -i '16,218d' CMakeLists.txt
+    sed -i '16iinclude_directories(${luajit}/include)' CMakeLists.txt
+    sed -i '17ilink_directories(${luajit}/lib)' CMakeLists.txt
+  '';
+
+  configurePhase = ''
+    export FORCE_GIT_VERSION=${version}
+    mkdir build-dir
+    cd build-dir
+    cmake -DCMAKE_INSTALL_PREFIX=$out ..
```

Comment:
When you add cmake, Nixpkgs already takes care of this.
Just put the `export ...` in a `preConfigure`.

Reply:
I'm doing this because the package doesn't build in either the root or the `build` folder. How to specify the CMake build folder in Nix?

Reply:
Understood.